<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>937</width>
    <height>736</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>2000</width>
    <height>2000</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>651</width>
      <height>51</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLineEdit" name="show_filePath"/>
     </item>
     <item>
      <widget class="QPushButton" name="btn_openFile">
       <property name="text">
        <string>打开文件</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_binary">
       <property name="text">
        <string>二值化</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_canny">
       <property name="text">
        <string>canny边缘提取</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_gauss">
       <property name="text">
        <string>Gauss模糊</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="widget_2" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>2001</width>
      <height>621</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>0</height>
     </size>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <widget class="QLabel" name="label_src">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>360</width>
       <height>270</height>
      </rect>
     </property>
     <property name="minimumSize">
      <size>
       <width>360</width>
       <height>270</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>360</width>
       <height>270</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="textFormat">
      <enum>Qt::AutoText</enum>
     </property>
     <property name="scaledContents">
      <bool>false</bool>
     </property>
     <property name="wordWrap">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_textCanny">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>290</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>原始图片</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_textBinary">
     <property name="geometry">
      <rect>
       <x>650</x>
       <y>290</y>
       <width>54</width>
       <height>12</height>
      </rect>
     </property>
     <property name="text">
      <string>二值化</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_canny">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>310</y>
       <width>360</width>
       <height>270</height>
      </rect>
     </property>
     <property name="minimumSize">
      <size>
       <width>360</width>
       <height>270</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>360</width>
       <height>270</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="text_draw">
     <property name="geometry">
      <rect>
       <x>630</x>
       <y>590</y>
       <width>54</width>
       <height>12</height>
      </rect>
     </property>
     <property name="text">
      <string>gauss模糊</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_draw">
     <property name="geometry">
      <rect>
       <x>480</x>
       <y>310</y>
       <width>360</width>
       <height>270</height>
      </rect>
     </property>
     <property name="minimumSize">
      <size>
       <width>360</width>
       <height>270</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>360</width>
       <height>270</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QSlider" name="slider_cannyLowValue">
     <property name="geometry">
      <rect>
       <x>390</x>
       <y>339</y>
       <width>22</width>
       <height>211</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>100</number>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="tickPosition">
      <enum>QSlider::TicksAbove</enum>
     </property>
    </widget>
    <widget class="QSlider" name="slider_gaussKernalSize">
     <property name="geometry">
      <rect>
       <x>870</x>
       <y>330</y>
       <width>22</width>
       <height>221</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>4</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="tickPosition">
      <enum>QSlider::TicksAbove</enum>
     </property>
    </widget>
    <widget class="QSlider" name="slider_cannyHighValue">
     <property name="geometry">
      <rect>
       <x>430</x>
       <y>340</y>
       <width>22</width>
       <height>211</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>200</number>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="tickPosition">
      <enum>QSlider::TicksAbove</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>380</x>
       <y>560</y>
       <width>54</width>
       <height>12</height>
      </rect>
     </property>
     <property name="text">
      <string>低阈值</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>420</x>
       <y>560</y>
       <width>54</width>
       <height>12</height>
      </rect>
     </property>
     <property name="text">
      <string>高阈值</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>860</x>
       <y>570</y>
       <width>54</width>
       <height>12</height>
      </rect>
     </property>
     <property name="text">
      <string>核大小</string>
     </property>
    </widget>
    <widget class="QLabel" name="text_cannyLowValue">
     <property name="geometry">
      <rect>
       <x>380</x>
       <y>310</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>100</string>
     </property>
     <property name="textFormat">
      <enum>Qt::PlainText</enum>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="text_cannyHighValue">
     <property name="geometry">
      <rect>
       <x>430</x>
       <y>310</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>200</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="text_gaussKernalSize">
     <property name="geometry">
      <rect>
       <x>860</x>
       <y>310</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>3</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="text_binaryValue">
     <property name="geometry">
      <rect>
       <x>880</x>
       <y>10</y>
       <width>18</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>100</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>860</x>
       <y>260</y>
       <width>60</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>二值化阈值</string>
     </property>
    </widget>
    <widget class="QSlider" name="slider_binaryValue">
     <property name="geometry">
      <rect>
       <x>880</x>
       <y>40</y>
       <width>27</width>
       <height>211</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>100</number>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="invertedAppearance">
      <bool>false</bool>
     </property>
     <property name="invertedControls">
      <bool>false</bool>
     </property>
     <property name="tickPosition">
      <enum>QSlider::TicksAbove</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_binary">
     <property name="geometry">
      <rect>
       <x>480</x>
       <y>10</y>
       <width>360</width>
       <height>270</height>
      </rect>
     </property>
     <property name="minimumSize">
      <size>
       <width>360</width>
       <height>270</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>360</width>
       <height>270</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <zorder>label_binary</zorder>
    <zorder>label_src</zorder>
    <zorder>label_textCanny</zorder>
    <zorder>label_textBinary</zorder>
    <zorder>label_canny</zorder>
    <zorder>text_draw</zorder>
    <zorder>label_draw</zorder>
    <zorder>slider_cannyLowValue</zorder>
    <zorder>slider_gaussKernalSize</zorder>
    <zorder>slider_cannyHighValue</zorder>
    <zorder>label_2</zorder>
    <zorder>label_3</zorder>
    <zorder>label_4</zorder>
    <zorder>text_cannyLowValue</zorder>
    <zorder>text_cannyHighValue</zorder>
    <zorder>text_gaussKernalSize</zorder>
    <zorder>text_binaryValue</zorder>
    <zorder>label</zorder>
    <zorder>slider_binaryValue</zorder>
   </widget>
   <widget class="QLabel" name="label_textContours_2">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>670</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>canny</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>937</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
